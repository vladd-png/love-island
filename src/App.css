.App {
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.input-form {
  background-color: #fff;
  padding: 5px;
  width: 40%;
  margin: 20px;
  border-radius: 50px;
  border: 15px dotted gold;
}

.love-card {
  background-color: #fff;
  border: indianred 5px solid;
  border-radius: 15px;
  width: 300px;
  margin: 10px;
}

.card-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.player-btn,
.add-btn {
  width: 100px;
  height: 30px;
  color: #fff;
  background-color: orange;
  margin: 5px;
  border-radius: 5px;
  font-size: 1em;
}

input {
  width: 200px;
  font-size: 1.2em;
  height: 30px;
}

h3 {
  font-size: 1.5em;
}

.add-btn {
  width: 300px;
  height: 50px;
  font-size: 1.1em;
}
