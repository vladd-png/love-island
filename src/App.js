import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import Form from './Form';
import CardContainer from './CardContainer.js';

class App extends Component {
  constructor() {
    super();
    this.state = {
      contestants: [
        { id: 1, name: 'Leta', status: 'online', interests: 'Cats and Burritos' }
      ]
    }
  }
  componentDidMount = () => {
    fetch('http://localhost:3001/api/v1/users')
      .then(response => response.json())
      .then(data => {
        this.setState({ contestants: data })
      })
  }
  addContestant = player => {
    this.setState({ contestants: [...this.state.contestants, player] })
  }
  addHeart = id => {
    let findHearted = this.state.contestants.filter(player => player.id === id);
    console.log(findHearted);
  }
  removeContestant = id => {
    const removePlayer = {
      method: 'DELETE',
      body: JSON.stringify({
        id: id
      }),
      headers: {
        'Content-type': 'application/json'
      }
    }
    fetch(`http://localhost:3001/api/v1/users/${id}`, removePlayer)
      .then(response => response.json())
      .then(data => {
        this.setState({ contestants: data })
      })
  }
  render() {
    return (
      <div className="App">
        <Form addContestant={this.addContestant}/>
        <CardContainer contestants={this.state.contestants} removeContestant={this.removeContestant} addHeart={this.addHeart} />
      </div>
    )
  }
}

export default App;
