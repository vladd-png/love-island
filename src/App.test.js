import React from 'react';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {

  it('should be an instance of App component', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should update its state when addContestant is called', () => {
    const player = {id: 2, name: 'Ranger', status: 'Busy', interests: 'Peanut Butter and Squirrels' }
    const expected = [
      {id: 1, name: 'Leta', status: 'online', interests: 'Cats and Burritos' },
      {id: 2, name: 'Ranger', status: 'Busy', interests: 'Peanut Butter and Squirrels' }
  ]
    const wrapper = shallow(<App />);
    wrapper.instance().addContestant(player);
    expect(wrapper.state('contestants')).toEqual(expected);
  })

});
