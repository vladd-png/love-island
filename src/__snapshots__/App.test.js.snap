// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should be an instance of App component 1`] = `
<div
  className="App"
>
  <Form
    addContestant={[Function]}
  />
  <CardContainer
    addHeart={[Function]}
    contestants={
      Array [
        Object {
          "id": 1,
          "interests": "Cats and Burritos",
          "name": "Leta",
          "status": "online",
        },
      ]
    }
    removeContestant={[Function]}
  />
</div>
`;
